examples/04_glm_first_level/plot_hrf.py

```diff
@@ -1,65 +1,163 @@
-the user. Fortunately, using the spm or Glover model does not make a huge
-difference.  Adding derivatives should be considered whenever timing
-information has some degree of uncertainty. It is actually useful to detect
+In current analysis frameworks, the choice of :term:`HRF` model is essentially
+left to the user. Fortunately, using the :term:`SPM` or
+Glover model does not make a huge
+difference. Adding derivatives should be considered whenever timing
+information has some degree of uncertainty, and is actually useful to detect
 timing issues.
 
+This example requires matplotlib and scipy.
 """
+
 #########################################################################
-# Set up some parameters for model display
-# ----------------------------------------
+# Define stimulus parameters and response models
+# ----------------------------------------------
 #
 # To get an impulse response, we simulate a single event occurring at time t=0,
 # with duration 1s.
 
 import numpy as np
-frame_times = np.linspace(0, 30, 61)
-onset, amplitude, duration = 0., 1., 1.
+
+time_length = 30.0
+frame_times = np.linspace(0, time_length, 61)
+onset, amplitude, duration = 0.0, 1.0, 1.0
 exp_condition = np.array((onset, duration, amplitude)).reshape(3, 1)
 
 #########################################################################
-# We sample this on a fris for display
+# Make a time array of this condition for display:
 stim = np.zeros_like(frame_times)
 stim[(frame_times > onset) * (frame_times <= onset + duration)] = amplitude
 
 #########################################################################
-# Now we have to define the candidate hrf models.
-hrf_models = [None, 'glover + derivative', 'glover + derivative + dispersion']
+# Define custom response functions for MION. Custom response
+# functions should at least take tr and oversampling as arguments:
+from scipy.stats import gamma
+
+
+def mion_response_function(tr, oversampling=16, onset=0.0):
+    """Implementation of the MION response function model.
+    Parameters
+    ----------
+    tr: float
+        scan repeat time, in seconds
+    oversampling: int, optional
+        temporal oversampling factor
+    onset: float, optional
+        hrf onset time, in seconds
+
+    Returns
+    -------
+    response_function: array of shape(length / tr * oversampling, dtype=float)
+        response_function sampling on the oversampled time grid
+    """
+    dt = tr / oversampling
+    time_stamps = np.linspace(
+        0, time_length, np.rint(float(time_length) / dt).astype(int)
+    )
+    time_stamps -= onset
+
+    # parameters of the gamma function
+    delay = 1.55
+    dispersion = 5.5
+
+    response_function = gamma.pdf(time_stamps, delay, loc=0, scale=dispersion)
+    response_function /= response_function.sum()
+    response_function *= -1
+
+    return response_function
+
+
+def mion_time_derivative(tr, oversampling=16.0):
```

Comment:
You don't want to put the derivative in the same function as the main mion reponse ?

Reply:
I just tried merging `mion_response_function` and `mion_time_derivative` into a single function returning an array, but `compute_regressor` fails to parse the output correctly and returns an error, so I don't think we can merge these functions.

I understand that you'd like the example to be as compact as possible, but I think this one is not too long and that example functions actually follow the same conventions as nilearn, which I think could be nice :relaxed: